GHzDACs/GHzAD_v7.md

```diff
@@ -0,0 +1,437 @@
+* Averaging Memory is halved to 8 us.
+* Can define now output data for packet.
+* FIFO is 8192 bytes; this is 186 ethernet packets.
+* Fixed ugly code, now Ethernet input is first latched.
+* Added counter for received packet and bad CRC packet.
+* Build = 7
+
+#### Version 6: Oct 2013-10-29
+
+* Revise version 2 to include 6 channels
+* Build = 6
+
+#### Version 5: Oct 2013-10-28
+
+* Revise version 3 to include 6 channels
+* Build = 5 
+
+#### Version 4: Aug. 2013
+
+* Add multiple triggering, fifo on output data, single bit output mode
+* Fifo output data, Daisy chain output from ADbits, enhanced monitors like DAC
+* Build = 4
+
+#### Version 3:  Do not use on prototype board (as FPGA pinouts have changed)
+
+* Add DCLK and outedge capability, as built into Board 1.0/not prototype
+* outedge=0 always.  DCLK phase in clockmon[1] output. 
+* Change LED[0] and LED[1] programming 
+* Build=3
+
+#### Version 2:
+
+* Add counter for sram trigger  
+* Add its readback register  
+* Build=2
+
+#### Version 1:
+
+* Only 4 (addresses 0 to 3) of the 11 demodulation channels are compiled
+* Build=1
+
+## SRAM Write
+
+### Genral description
+
+The controller must write to SRAM in the AD board to define two tables,
+a retrigger table to define multiple AD triggers for an experiment start,
+and a multiplier table for demodulation of the incoming signal for each demodulation channel.
+
+#### Retrigger table
+
+The retrigger table defines multiple AD triggers per master start.
+The table starts at address 0, then increments up (to a maximum 127) after performing the functions of each table entry,
+until it reaches a table entry with `rdelay[15..8]=0`, at which time the retriggering stops. 
+Note that an empty table with `rdelay[15..9]=0` at the address 0 will not ever trigger the AD.
+In the table entry, `rdelay[15..0] + 3` is the number of 4 ns cycles between the AD start (or last end of ADon) and the AD trigger.
+After this delay, the AD is turned on for `rlength[7..0] + 1` cycles, during which ADon demultiplexer (don) is high.
+The value `rcount[15..0] + 1` is the number of AD triggers per table entry.
+An AD done signal pulses 3 cycles after the last ADon goes low.
+Note that there is approximately 7 clock cycle delay that needs to be measured and taken into account to get demodulation time correct.
+Channels 0 to `rchan[3..0] - 1` is read out; maximum `rchan[3..0]` is 11 for 12 channels.
+If `rchan[3..0] = 0`, then save data in bit readout mode.
+See below.
+
+Note that you have multiple triggers, as for the DAC board.
+But for each trigger, you can have multiple retriggering to account for multiple AD demodulations during each sequence.  
+
+#### Mixer table
+
+The multiplier sram table is now 8 bit integers that is multiplied to the input waveforms I and Q.
+This includes sine and cosine parts of the demodulation signal, along with any waveform shaping that should go to zero at beginning and end to minimize aliasing.
```

Comment:
A simple demodulator table would be W(t)*e^(-2j*pi*f*t) where f is the demoduation frequency and W(t) is a standard real-valued window function that goes to zero at the ends to prevent spectral leakage.  However, any complex function may be used here.